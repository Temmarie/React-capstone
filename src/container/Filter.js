import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { filterPokemons } from '../actions/index';
import details from '../css/Details.module.css';

const Filter = props => {
  const handleChange = e => {
    const { filterPokemons } = props;
    filterPokemons(e.target.value);
  };

  return (
    <div>
      <input
        type="text"
        onChange={handleChange}
        placeholder="Search Pokemon.."
        className={details.filter}
      />
    </div>
  );
};

Filter.propTypes = {
  filterPokemons: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => bindActionCreators({
  filterPokemons,
}, dispatch);

export default connect(null, mapDispatchToProps)(Filter);
