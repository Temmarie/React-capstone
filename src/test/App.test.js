import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import Filter from '../container/Filter';

const state = {
  pokemons: [{ name: 'pikachu' }],
  pokemon: {},
};
const mockStore = configureMockStore(state);

describe('Should Render Filter Component Properly', () => {
  describe('Rendering Filter component', () => {
    const store = mockStore();
    const component = renderer.create(<Provider store={store}><Filter /></Provider>);
    it('should render Filter component', () => {
      expect(component.toJSON).toMatchSnapshot();
    });
  });
});
