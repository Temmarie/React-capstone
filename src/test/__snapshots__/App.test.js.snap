// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render Filter Component Properly Rendering Filter component should render Filter component 1`] = `[Function]`;
